src/Marten/ISerializer.cs

```diff
@@ -7,6 +7,13 @@ namespace Marten
     public interface ISerializer
     {
         /// <summary>
+        /// Serialize the document object into <paramref name="writer"/>.
+        /// </summary>
+        /// <param name="document"></param>
+        /// <param name="writer"></param>
+        void ToJson(object document, TextWriter writer);
```

Comment:
@jeremydmiller Can we change the seam? From semver perspective this is going to be a breaking change (major). We could provide a new interface that would be implemented by serializers. For now I assume it's ok to change this.